src/afl-fuzz-stats.c

```diff
@@ -758,25 +758,27 @@ void show_stats(afl_state_t *afl) {
   SAYF(bV bSTOP " last uniq crash : " cRST "%-33s " bSTG bV bSTOP
                 " uniq crashes : %s%-6s" bSTG               bV "\n",
        time_tmp, crash_color, tmp);
-
-  sprintf(tmp, "%s%s", u_stringify_int(IB(0), afl->unique_hangs),
-          (afl->unique_hangs >= KEEP_UNIQUE_HANG) ? "+" : "");
-
   if (afl->schedule >= FAST && afl->schedule <= RARE) {
 
-    u64 is_one = 0;
-    for (u32 i = 0; i < N_FUZZ_SIZE; ++i) {
+    u64 singleton = 0;
+    for (u32 i = 0; i < afl->queued_paths; i++) {
 
-      if (afl->n_fuzz_tmp[i] == 1) { ++is_one; }
+      struct queue_entry *q = afl->queue_buf[i];
+      if (afl->n_fuzz_tmp[q->n_fuzz_entry] == 1) { ++singleton; }
 
     }
-
-    if (is_one) {
-
-      u64 t2f = (afl->fsrv.total_execs *
-                 (afl->total_cal_us / afl->total_cal_cycles)) /
-                is_one;
-      u_stringify_time_diff(time_tmp, t2f * 1000, 1);
+    sprintf(tmp, "%s%s", u_stringify_int(IB(0), singleton),
+          (afl->unique_hangs >= KEEP_UNIQUE_HANG) ? "+" : "");
+  } else {
+  sprintf(tmp, "%s%s", u_stringify_int(IB(0), afl->unique_hangs),
+          (afl->unique_hangs >= KEEP_UNIQUE_HANG) ? "+" : "");
+  }
+
+  if (afl->schedule >= FAST && afl->schedule <= RARE) {
+
+    if (afl->time_to_next != 0xFFFFFFFF) {
+
+      u_stringify_time_diff(time_tmp, afl->time_to_next, 1);

```

Comment:
What about using the following here?

```
u_stringify_time_diff(time_tmp, afl->last_path_time + afl->time_to_next, cur_ms);
```

Reply:
If we are over-estimating this makes sense. Otherwise, we could be negative pretty often.

Reply:
Yeah, I haven't tried it yet. I think for testing it would be good to see how often it becomes negative. Once we're happy we can turn the negative number into something more optimistic like "any second now". ;)